geth/jail/console/console_test.go

```diff
@@ -0,0 +1,97 @@
+package console_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/robertkrimen/otto"
+	"github.com/status-im/status-go/geth/jail/console"
+	"github.com/status-im/status-go/geth/node"
+	"github.com/stretchr/testify/suite"
+)
+
+// TestConsole validates the behaviour of giving conole extensions.
+func TestConsole(t *testing.T) {
+	suite.Run(t, new(ConsoleTestSuite))
+}
+
+type ConsoleTestSuite struct {
+	suite.Suite
+	vm *otto.Otto
+}
+
+func (s *ConsoleTestSuite) SetupTest() {
+	vm := otto.New()
+	require := s.Require()
+
+	require.NotNil(vm)
+	require.IsType(&otto.Otto{}, vm)
+
+	s.vm = vm
+}
+
+// TestConsoleLog will validate the operations of the console.log extension
+// for the otto vm.
+func (s *ConsoleTestSuite) TestConsoleLog() {
+	require := s.Require()
+	written := "Bob Marley"
+
+	var customWriter bytes.Buffer
+
+	// register console handler.
+	err := s.vm.Set("console", map[string]interface{}{
+		"log": console.Extension(&customWriter, console.Log),
+	})
+	require.NoError(err)
+
+	_, err = s.vm.Run(fmt.Sprintf(`console.log(%q);`, written))
+	require.NoError(err)
+
+	require.NotEmpty(&customWriter)
```

Comment:
This line might be also obsolete as the next `require.Equal()` validates the same, but even more strictly. Unless there is some gotcha with `require`.